src/core/Akka/IO/SocketEventArgsPool.cs

```diff
@@ -135,7 +135,10 @@ public static void SetBuffer(this SocketAsyncEventArgs args, IEnumerable<ByteStr
             }
             else
             {
-                args.SetBuffer(null, 0, 0);
+                if (args.Buffer != null)
```

Comment:
SetBuffer(Null) has a lot going on in it. Checking if the buffer is null, on the other hand, is a simple Get in `SocketAsyncEventArgs`